Android 13 migration update compile and target sdk versions to 33

ravishanker opened the pull request:
Updates compileSdkVersion and targetSdkVersion from 31 to 33 (Android 13)

Internal project thread: pbArwn-5IN-p2

Fixes: [#18096](https://github.com/wordpress-mobile/WordPress-Android/issues/18096)

To test

- Checkout this branch
- Use this branch through WP (uncomment localStoriesAndroidPath in local-builds.gradle)
- Try out adding a Story post (FAB > Story post) with an image and/or video
- Ensure, no issues or crashes

ravishanker committed:
Update compileSdkVersion to 33

ravishanker committed:
Update Mp4ComposerBasic.java

Catch Exception

ravishanker committed:
Update onFling method signature

ravishanker commented:
@ParaskP7 - Looks like stories-android still has CircleCI checks!

ParaskP7 commented:
👋 @ravishanker !

> @ParaskP7 - Looks like stories-android still has CircleCI checks!

You're right, it looks that the `CircleCI` to `Buildkite` migration hasn't been completed for this repo, just like it happened for the rest of them, and to be honest, I am not sure why... 🤔 Cc @oguzkocer

oguzkocer commented:
@jkmassel was working on these migrations, but we have a lot of repos, so it's most likely an oversight. If this is a blocker, I can try and handle this migration this week.

ravishanker commented:
> @jkmassel was working on these migrations, but we have a lot of repos, so it's most likely an oversight. If this is a blocker, I can try and handle this migration this week.

@oguzkocer Yes, please.  Thank you

oguzkocer commented:
> @oguzkocer Yes, please. Thank you

I've moved the lint & unit test tasks to Buildkite in #729. I'm waiting some feedback from the App Infrastructure Team about the installable build, but worst case scenario, we can merge #729 tomorrow and open a separate PR for the installable build.

@ravishanker If you can let me know the urgency of this PR, I can take the appropriate action. Thank you!

irfano commented:
> > @oguzkocer Yes, please. Thank you
> 
> I've moved the lint & unit test tasks to Buildkite in #729. I'm waiting some feedback from the App Infrastructure Team about the installable build, but worst case scenario, we can merge #729 tomorrow and open a separate PR for the installable build.
> 
> @ravishanker If you can let me know the urgency of this PR, I can take the appropriate action. Thank you!

Thank you for handling this, @oguzkocer! This is not too urgent and does not need to be merged this week. It can be merged in our next sprint.

oguzkocer commented:
Thanks @irfano! https://github.com/Automattic/stories-android/pull/729 is now ready for review, so it should be merged early next week 🤞

oguzkocer commented:
@ravishanker @irfano https://github.com/Automattic/stories-android/pull/729 is merged. You can merge `trunk` into this branch which should trigger Buildkite to run the jobs instead of CircleCI. Please let me know if you have any issues!

ravishanker committed:
Merge branch 'trunk' into Android-13-migration-update-compile-sdk-version-to-33

ravishanker committed:
Update PhotoPickerAdapter.java

add try, catch around MediaMetadataRetriever release

ravishanker committed:
Update targetSdk to 33

Update targetSdkVersion from 31 to 33

irfano commented on build.gradle:
```diff
@@ -20,7 +20,7 @@ allprojects {
             }
         }
         google()
-        jcenter()
```
Removing jcenter causes failure of fetching some libraries for the example app. (for example: `com.facebook.flipper:0.50.0`)
I think we can revert this change to avoid resolving more errors on the example app.

ravishanker replied:
Looks like[ flipper is available from 0.54.0 on mavenCentral](https://mvnrepository.com/artifact/com.facebook.flipper/flipper).  Updated the flipper version to that is on Automattic version catalog.  All failures have disappeared now.

irfano replied:
It's fixed! 👍🏻

irfano reviewed (changes requested):
I have requested a minor change. Once it is resolved, we will be ready to merge.

ravishanker committed:
Remove jcenter

Replace jcenter with mavenCentral and update flipper version with version catalog

irfano reviewed (changes requested):
I tested the Loop demo app but it isn't going to the next screen after requesting permission. I think the permission code needs to be updated.

ravishanker commented:
> I tested the Loop demo app but it isn't going to the next screen after requesting permission. I think the permission code needs to be updated.

It has READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE .  Better handle that separately as part of [media permissions task](https://github.com/wordpress-mobile/WordPress-Android/issues/17715)

ravishanker committed:
Update permissions

Remove READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE

Add READ_MEDIA_AUDIO, READ_MEDIA_IMAGES and READ_MEDIA_AUDIO

irfano reviewed (approved):
@ravishanker 👋🏻 
`READ_EXTERNAL_STORAGE` permission still exists in `photoeditor` package. But I'm not sure if it's required or not. If you like to investigate it to ensure it won't break anything, you can do it before merging. Otherwise feel free to merge it. I'm approving the PR. It works on WPAndroid app and also on the example.

ravishanker committed:
Update PhotoEditor permission

Replace READ and WRITE to EXTERNAL STORAGE with READ MEDA permissions
